model/task/task_test.go

```diff
@@ -3140,6 +3141,124 @@ func TestArchiveMany(t *testing.T) {
 	}
 }
 
+func TestArchiveManyFailedOnly(t *testing.T) {
+	require.NoError(t, db.ClearCollections(Collection, OldCollection))
+	t1_et1 := Task{
+		Id:        "t1_et1",
+		Status:    evergreen.TaskFailed,
+		Execution: 2,
+		Aborted:   true,
+		Version:   "v",
+	}
+	assert.NoError(t, t1_et1.Insert())
+	t1_et2 := Task{
+		Id:        "t1_et2",
+		Status:    evergreen.TaskSucceeded,
+		Execution: 2,
+		Aborted:   true,
+		Version:   "v",
+	}
+	assert.NoError(t, t1_et2.Insert())
+	t1 := Task{
+		Id:             "t1",
+		Status:         evergreen.TaskFailed,
+		ExecutionTasks: []string{t1_et1.Id, t1_et2.Id},
+		Execution:      2,
+		Aborted:        true,
+		DisplayOnly:    true,
+		Version:        "v",
+	}
+	assert.NoError(t, t1.Insert())
+	t2 := Task{
+		Id:        "t2",
+		Status:    evergreen.TaskSucceeded,
+		Execution: 3,
+		Aborted:   true,
+		Version:   "v",
+	}
+	assert.NoError(t, t2.Insert())
+	t3_et1 := Task{
+		Id:        "t3_et1",
+		Status:    evergreen.TaskFailed,
+		Execution: 0,
+		Aborted:   true,
+		Version:   "v",
+	}
+	assert.NoError(t, t3_et1.Insert())
+	t3_et2 := Task{
+		Id:        "t3_et2",
+		Status:    evergreen.TaskSucceeded,
+		Execution: 0,
+		Aborted:   true,
+		Version:   "v",
+	}
+	assert.NoError(t, t3_et2.Insert())
+	t3_et3 := Task{
+		Id:        "t3_et3",
+		Status:    evergreen.TaskFailed,
+		Execution: 0,
+		Aborted:   true,
+		Version:   "v",
+	}
+	assert.NoError(t, t3_et3.Insert())
+	t3 := Task{
+		Id:                      "t3",
+		Status:                  evergreen.TaskFailed,
+		ExecutionTasks:          []string{t3_et1.Id, t3_et2.Id, t3_et3.Id},
+		Execution:               0,
+		Aborted:                 true,
+		DisplayOnly:             true,
+		ResetFailedWhenFinished: true,
+		Version:                 "v",
+	}
+	assert.NoError(t, t3.Insert())
+	assert.NoError(t, t3.Archive()) // Failed only is true
```

Comment:
Should we check the results of having archived t3 with failed only true? (i.e. that just some exec tasks are restarted, and that LPE is what we expect?)